src/test/java/org/cactoos/iterable/ReversedTest.java

```diff
@@ -23,32 +23,102 @@
  */
 package org.cactoos.iterable;
 
-import org.cactoos.text.Joined;
-import org.hamcrest.MatcherAssert;
+import org.cactoos.list.ListOf;
+import org.hamcrest.collection.IsEmptyIterable;
+import org.hamcrest.core.IsEqual;
+import org.hamcrest.core.IsNot;
 import org.junit.Test;
-import org.llorllale.cactoos.matchers.TextHasString;
+import org.llorllale.cactoos.matchers.Assertion;
+import org.llorllale.cactoos.matchers.HasSize;
 
 /**
  * Test case for {@link Reversed}.
  * @since 0.9
  * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
  */
 public final class ReversedTest {
 
     @Test
     public void reversesIterable() {
-        MatcherAssert.assertThat(
-            "Can't reverse an iterable",
-            new Joined(
-                " ",
-                new Reversed<>(
-                    new IterableOf<>(
-                        "hello", "world", "dude"
-                    )
+        new Assertion<>(
+            "Must reverse an iterable",
+            new Reversed<>(
+                new IterableOf<>(
+                    "h", "w", "d"
                 )
             ),
-            new TextHasString("dude world hello")
+            new IsEqual<>(new IterableOf<>("d", "w", "h"))
+        ).affirm();
+    }
+
+    @Test
+    public void iteratesMultipleTimes() {
```

Comment:
@victornoel I didn't understood this test, seems that nothing is being done... Also, could you try to put it in just one assertion?

Reply:
@paulodamaso I wanted to verify that that the iterable could be iterated multiple times, I will transform it so that it's clearer and also with only one assertion

Reply:
@victornoel Well, I think that it was better the other way, could you please change it back?

Reply:
@paulodamaso hehe, no problem, it's done :)